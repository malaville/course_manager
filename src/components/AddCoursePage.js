import React from "react";

export default class AddCoursePage extends React.Component {
  state = {
    title: "",
    short_name: "",
    main_teacher: "",
    description: ""
  };

  onInputChange = e => {
    const value = e.target.value;
    const field = e.target.name;
    this.setState(() => ({
      [field]: value
    }));
  };

  render() {
    return (
      <div>
        This is from my add expense component
        <form>
          Title and short name : <br />
          <input
            type="text"
            placeholder="Title of the course"
            name="title"
            value={this.state.title}
            onChange={e => this.onInputChange(e)}
          />
          <input
            type="text"
            placeholder="Short name (i.e. CR3)"
            name="short_name"
            value={this.state.short_name}
            onChange={e => this.onInputChange(e)}
          />
          <br />
          Professor in charge :
          <br />
          <input
            type="text"
            placeholder="Professor in charge"
            name="main_teacher"
            value={this.state.main_teacher}
            onChange={e => this.onInputChange(e)}
          />
          <br />
          Description : <br />
          <textarea
            type="text"
            rows="10"
            placeholder="Description"
            name="description"
            value={this.state.description}
            onChange={e => this.onInputChange(e)}
          />
          <br />
          <button type="submit">Submit</button>
        </form>
      </div>
    );
  }
}

// const mock_course2 = {
//   id: 2,
//   title: "Management de l'innovation ",
//   short_name: "I4",
//   main_teacher: "J-M. Dalle",
//   description: "This course is the first of the innovation seminary",
//   last_modified: 0,
//   lessons: [
//     {
//       location: "Telecom",
//       hour: "16:30",
//       date: "2019-01-18",
//       course_id: 2,
//       lesson_number: 1,
//       description:
//         "M. Bry will give you very interesting tips to build your first business model !"
//     },
//     {
//       location: "Telecom",
//       hour: "16:30",
//       date: "2019-01-27",
//       course_id: 2,
//       lesson_number: 2,
//       description:
//         "Nicolas Bry will introduce you to the famous post-it technique"
//     },
//     {
//       location: "Telecom",
//       hour: "14:30",
//       date: "2019-01-26",
//       course_id: 2,
//       lesson_number: -1,
//       description:
//         "You will now present your project, even if it is very shitty"
//     },
//     {
//       location: "Telecom",
//       hour: "9:30",
//       date: "2019-02-12",
//       course_id: 2,
//       lesson_number: -1,
//       description: "Ce cours n'existe pas"
//     }
//   ]
// };
