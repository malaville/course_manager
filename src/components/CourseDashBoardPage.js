import React from "react";

const CourseDashBoardPage = () => <div>This Page shows your courses</div>;

export default CourseDashBoardPage;
