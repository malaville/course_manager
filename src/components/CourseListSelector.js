import React from "react";
import { connect } from "react-redux";
import { setCourseIdFilters, sortLessonsBy } from "../actions/userInterface";

const CourseItemCheckbox = ({ short_name, id, dispatch }) => (
  <label>
    <input
      type="checkbox"
      id={id}
      name={short_name}
      onChange={e => {
        dispatch(setCourseIdFilters(e.target.id, e.target.checked));
        // OODODO
      }}
    />
    {short_name}
  </label>
);

const SortingSelector = ({ filters, dispatch, selected_id }) => (
  <select
    defaultValue={selected_id}
    onChange={e => dispatch(sortLessonsBy(e.target.value))}
  >
    {filters.map(filter => (
      <option key={filter.id} value={filter.id}>
        {filter.name}
      </option>
    ))}
  </select>
);

const CourseListSelector = props => {
  const SelectorParams = {
    filters: [
      { name: "date", id: "date" },
      { name: "course", id: "main_course_date" }
    ],
    selected_id: "main_course_date" //props.lessonFilter
  };
  return (
    <form>
      {props.courses.map(course => (
        <CourseItemCheckbox
          key={course.id}
          dispatch={props.dispatch}
          {...course}
        />
      ))}
      <SortingSelector dispatch={props.dispatch} {...SelectorParams} />
    </form>
  );
};

const mapStateToProps = state => {
  return {
    courses: state.courses,
    lessonFilter: state.userInterface.sortLessonsBy
  };
};

export default connect(mapStateToProps)(CourseListSelector);
