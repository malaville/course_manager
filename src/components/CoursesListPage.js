import React from "react";
import { Redirect } from "react-router-dom";

const coursesList = [["1", "The First Course"], ["2", "The Second Course"]];

const CoursesListPage = props => {
  console.log(props);

  let courseIsInList = !props.match.params.id || false;
  let names_are_not_matching = false;
  let url_name;

  coursesList.forEach((course, index) => {
    if (course[0] == props.match.params.id) {
      courseIsInList = true;
      const number_course_in_list = index;
      const name_of_the_course = coursesList[number_course_in_list][1];
      url_name = name_of_the_course.toLowerCase().replace(/\s+/g, "_");
      names_are_not_matching =
        !props.match.params.name_of_the_course ||
        url_name != props.match.params.name_of_the_course;
    }
  });

  if (!courseIsInList) {
    return <Redirect to="/courses/" />;
  }

  if (names_are_not_matching) {
    return <Redirect to={`/courses/${props.match.params.id}/${url_name}`} />;
  }

  return (
    <div>
      This is from my edit expense component.
      <br />
      {props.match.params.name_of_the_course &&
        `You queried the course nÂ°${
          props.match.params.id
        } and name ${props.match.params.name_of_the_course
          .toUpperCase()
          .replace(/_/g, " ")}`}
    </div>
  );
};

export default CoursesListPage;
