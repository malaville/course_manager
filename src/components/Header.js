import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Course Manager</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>
      Dashboard
    </NavLink>
    <NavLink to="/create" activeClassName="is-active">
      Create Course
    </NavLink>
    <NavLink to="/courses" activeClassName="is-active">
      Courses List
    </NavLink>
    <NavLink to="/help" activeClassName="is-active">
      Help
    </NavLink>
  </header>
);

export default Header;
