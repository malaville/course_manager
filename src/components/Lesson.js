import React from "react";
const Lesson = props => {
  return (
    <li key={props.key}>
      <h4>
        {"Séance°" +
          props.lesson_number +
          " - " +
          props.date +
          " " +
          props.hour}
      </h4>
      <p>
        Location : {props.location}
        <br />
        Description : {props.description}
      </p>
    </li>
  );
};

export default Lesson;
