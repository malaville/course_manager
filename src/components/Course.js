import React from "react";
import Lesson from "./Lesson";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { addLesson } from "../actions/courses";

const Course = ({
  short_name,
  title,
  id,
  main_teacher,
  description,
  lessons,
  showLessons,
  dispatch
}) => {
  return (
    <li key={short_name}>
      <h3>
        {short_name + " - " + title}{" "}
        <Link to={`/courses/${id}/${title.toLowerCase().replace(/\s+/g, "_")}`}>
          {" "}
          d√©tails
        </Link>
      </h3>
      <p>
        Main Teacher : {main_teacher}
        <br />
        Description : {description}
        <br />
      </p>
      {showLessons && (
        <ul>
          {lessons.map(lesson => (
            <Lesson
              key={lesson.course_id + "_" + lesson.lesson_number}
              {...lesson}
              course_short_name={short_name}
              editable
            />
          ))}
          <li>
            <button
              className="btn btn-link"
              onClick={() => dispatch(addLesson(id))}
            >
              Ajouter un cours
            </button>
          </li>
        </ul>
      )}
    </li>
  );
};

export default connect()(Course);
