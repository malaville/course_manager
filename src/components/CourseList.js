import React from 'react';
import { connect } from 'react-redux';
import Course from './Course';

export const CourseList = props => {
  return (
    <ul>
      {props.focusId ? (
        <Course
          showLessons={true}
          {...props.courses.filter(course => {
            return course.id == props.focusId;
          })[0]}
        />
      ) : (
        props.courses.map(course => <Course key={course.id} {...course} showLessons={false} />)
      )}
    </ul>
  );
};
const mapStateToProps = state => ({
  courses: state.courses
});

export default connect(mapStateToProps)(CourseList);
