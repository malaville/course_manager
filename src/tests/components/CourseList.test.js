import { CourseList } from '../../components/CourseList';
import React from 'react';
import { shallow } from 'enzyme';
import { mock_course_list } from '../test-data/courses-lessons';
import Course from '../../components/Course';

test('The course list without focusId should render 2 courses and match snapshot, showLessons should be falsy on each because we do not show lessons in this case, ', () => {
  const shallowCourseList = shallow(<CourseList courses={mock_course_list} />);
  expect(shallowCourseList.find(Course).length).toBe(2);
  expect(
    shallowCourseList
      .find(Course)
      .first()
      .props().showLessons
  ).toBeFalsy();
  expect(shallowCourseList).toMatchSnapshot();
});

test('The course list WITH focusId should render 1 course, and match snapshot', () => {
  const shallowCourseList = shallow(<CourseList courses={mock_course_list} focusId={1} />);
  expect(shallowCourseList.find(Course).length).toBe(1);
  expect(
    shallowCourseList
      .find(Course)
      .first()
      .props().showLessons
  ).toBeTruthy();
  expect(shallowCourseList).toMatchSnapshot();
});
