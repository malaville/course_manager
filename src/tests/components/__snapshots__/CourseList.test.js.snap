// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The course list WITH focusId should render 1 course, and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourseList
    courses={
        Array [
            Object {
              "description": "This course is the third of the innovation seminary",
              "id": 1,
              "lessons": Array [
                Object {
                  "course_id": 1,
                  "course_short_name": "SEMI",
                  "date": "2019-01-18",
                  "description": "JMD will be talking about Agoranov for one long hour",
                  "hour": "16:45",
                  "lesson_number": 1,
                  "location": "Agoranov",
                },
                Object {
                  "course_id": 1,
                  "course_short_name": "SEMI",
                  "date": undefined,
                  "description": "JB Bengo will be talking about very interesting things",
                  "hour": "15:15",
                  "lesson_number": 2,
                  "location": "Telecom",
                },
                Object {
                  "course_id": 1,
                  "course_short_name": "SEMI",
                  "date": "2019-01-18",
                  "description": "JB Bengo will be talking about very interesting things",
                  "hour": "15:18",
                  "lesson_number": 3,
                  "location": "Telecom",
                },
                Object {
                  "course_id": 1,
                  "course_short_name": "SEMI",
                  "date": "2019-01-18",
                  "description": "Version2 : JB Bengo will be talking about very interesting things",
                  "hour": "15:18",
                  "lesson_number": 4,
                  "location": "Telecom",
                },
              ],
              "main_teacher": "J-M. Dalle",
              "short_name": "SEMI",
              "title": "Seminaire Innovation",
            },
            Object {
              "description": "This course is the first of the innovation seminary",
              "id": 2,
              "last_modified": 0,
              "lessons": Array [
                Object {
                  "course_id": 2,
                  "course_short_name": "I4",
                  "date": "2019-01-18",
                  "description": "M. Bry will give you very interesting tips to build your first business model !",
                  "hour": "16:30",
                  "lesson_number": 1,
                  "location": "Telecom",
                },
                Object {
                  "course_id": 2,
                  "course_short_name": "I4",
                  "date": "2019-01-27",
                  "description": "Nicolas Bry will introduce you to the famous post-it technique",
                  "hour": "16:30",
                  "lesson_number": 2,
                  "location": "Telecom",
                },
                Object {
                  "course_id": 2,
                  "course_short_name": "I4",
                  "date": "2019-01-26",
                  "description": "You will now present your project, even if it is very shitty",
                  "hour": "14:30",
                  "lesson_number": 3,
                  "location": "Telecom",
                },
                Object {
                  "course_id": 2,
                  "course_short_name": "I4",
                  "date": "",
                  "description": "Ce cours n'existe pas",
                  "hour": "9:30",
                  "lesson_number": 10,
                  "location": "Telecom",
                },
              ],
              "main_teacher": "J-M. Dalle",
              "short_name": "I4",
              "title": "Management de l'innovation ",
            },
          ]
    }
    focusId={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(Course)
        description="This course is the third of the innovation seminary"
        id={1}
        lessons={
                Array [
                        Object {
                          "course_id": 1,
                          "course_short_name": "SEMI",
                          "date": "2019-01-18",
                          "description": "JMD will be talking about Agoranov for one long hour",
                          "hour": "16:45",
                          "lesson_number": 1,
                          "location": "Agoranov",
                        },
                        Object {
                          "course_id": 1,
                          "course_short_name": "SEMI",
                          "date": undefined,
                          "description": "JB Bengo will be talking about very interesting things",
                          "hour": "15:15",
                          "lesson_number": 2,
                          "location": "Telecom",
                        },
                        Object {
                          "course_id": 1,
                          "course_short_name": "SEMI",
                          "date": "2019-01-18",
                          "description": "JB Bengo will be talking about very interesting things",
                          "hour": "15:18",
                          "lesson_number": 3,
                          "location": "Telecom",
                        },
                        Object {
                          "course_id": 1,
                          "course_short_name": "SEMI",
                          "date": "2019-01-18",
                          "description": "Version2 : JB Bengo will be talking about very interesting things",
                          "hour": "15:18",
                          "lesson_number": 4,
                          "location": "Telecom",
                        },
                      ]
        }
        main_teacher="J-M. Dalle"
        short_name="SEMI"
        showLessons={true}
        title="Seminaire Innovation"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "description": "This course is the third of the innovation seminary",
        "id": 1,
        "lessons": Array [
          Object {
            "course_id": 1,
            "course_short_name": "SEMI",
            "date": "2019-01-18",
            "description": "JMD will be talking about Agoranov for one long hour",
            "hour": "16:45",
            "lesson_number": 1,
            "location": "Agoranov",
          },
          Object {
            "course_id": 1,
            "course_short_name": "SEMI",
            "date": undefined,
            "description": "JB Bengo will be talking about very interesting things",
            "hour": "15:15",
            "lesson_number": 2,
            "location": "Telecom",
          },
          Object {
            "course_id": 1,
            "course_short_name": "SEMI",
            "date": "2019-01-18",
            "description": "JB Bengo will be talking about very interesting things",
            "hour": "15:18",
            "lesson_number": 3,
            "location": "Telecom",
          },
          Object {
            "course_id": 1,
            "course_short_name": "SEMI",
            "date": "2019-01-18",
            "description": "Version2 : JB Bengo will be talking about very interesting things",
            "hour": "15:18",
            "lesson_number": 4,
            "location": "Telecom",
          },
        ],
        "main_teacher": "J-M. Dalle",
        "short_name": "SEMI",
        "showLessons": true,
        "title": "Seminaire Innovation",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(Course)
          description="This course is the third of the innovation seminary"
          id={1}
          lessons={
                    Array [
                              Object {
                                "course_id": 1,
                                "course_short_name": "SEMI",
                                "date": "2019-01-18",
                                "description": "JMD will be talking about Agoranov for one long hour",
                                "hour": "16:45",
                                "lesson_number": 1,
                                "location": "Agoranov",
                              },
                              Object {
                                "course_id": 1,
                                "course_short_name": "SEMI",
                                "date": undefined,
                                "description": "JB Bengo will be talking about very interesting things",
                                "hour": "15:15",
                                "lesson_number": 2,
                                "location": "Telecom",
                              },
                              Object {
                                "course_id": 1,
                                "course_short_name": "SEMI",
                                "date": "2019-01-18",
                                "description": "JB Bengo will be talking about very interesting things",
                                "hour": "15:18",
                                "lesson_number": 3,
                                "location": "Telecom",
                              },
                              Object {
                                "course_id": 1,
                                "course_short_name": "SEMI",
                                "date": "2019-01-18",
                                "description": "Version2 : JB Bengo will be talking about very interesting things",
                                "hour": "15:18",
                                "lesson_number": 4,
                                "location": "Telecom",
                              },
                            ]
          }
          main_teacher="J-M. Dalle"
          short_name="SEMI"
          showLessons={true}
          title="Seminaire Innovation"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "description": "This course is the third of the innovation seminary",
          "id": 1,
          "lessons": Array [
            Object {
              "course_id": 1,
              "course_short_name": "SEMI",
              "date": "2019-01-18",
              "description": "JMD will be talking about Agoranov for one long hour",
              "hour": "16:45",
              "lesson_number": 1,
              "location": "Agoranov",
            },
            Object {
              "course_id": 1,
              "course_short_name": "SEMI",
              "date": undefined,
              "description": "JB Bengo will be talking about very interesting things",
              "hour": "15:15",
              "lesson_number": 2,
              "location": "Telecom",
            },
            Object {
              "course_id": 1,
              "course_short_name": "SEMI",
              "date": "2019-01-18",
              "description": "JB Bengo will be talking about very interesting things",
              "hour": "15:18",
              "lesson_number": 3,
              "location": "Telecom",
            },
            Object {
              "course_id": 1,
              "course_short_name": "SEMI",
              "date": "2019-01-18",
              "description": "Version2 : JB Bengo will be talking about very interesting things",
              "hour": "15:18",
              "lesson_number": 4,
              "location": "Telecom",
            },
          ],
          "main_teacher": "J-M. Dalle",
          "short_name": "SEMI",
          "showLessons": true,
          "title": "Seminaire Innovation",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`The course list without focusId should render 2 courses and match snapshot, showLessons should be falsy on each because we do not show lessons in this case,  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourseList
    courses={
        Array [
            Object {
              "description": "This course is the third of the innovation seminary",
              "id": 1,
              "lessons": Array [
                Object {
                  "course_id": 1,
                  "course_short_name": "SEMI",
                  "date": "2019-01-18",
                  "description": "JMD will be talking about Agoranov for one long hour",
                  "hour": "16:45",
                  "lesson_number": 1,
                  "location": "Agoranov",
                },
                Object {
                  "course_id": 1,
                  "course_short_name": "SEMI",
                  "date": undefined,
                  "description": "JB Bengo will be talking about very interesting things",
                  "hour": "15:15",
                  "lesson_number": 2,
                  "location": "Telecom",
                },
                Object {
                  "course_id": 1,
                  "course_short_name": "SEMI",
                  "date": "2019-01-18",
                  "description": "JB Bengo will be talking about very interesting things",
                  "hour": "15:18",
                  "lesson_number": 3,
                  "location": "Telecom",
                },
                Object {
                  "course_id": 1,
                  "course_short_name": "SEMI",
                  "date": "2019-01-18",
                  "description": "Version2 : JB Bengo will be talking about very interesting things",
                  "hour": "15:18",
                  "lesson_number": 4,
                  "location": "Telecom",
                },
              ],
              "main_teacher": "J-M. Dalle",
              "short_name": "SEMI",
              "title": "Seminaire Innovation",
            },
            Object {
              "description": "This course is the first of the innovation seminary",
              "id": 2,
              "last_modified": 0,
              "lessons": Array [
                Object {
                  "course_id": 2,
                  "course_short_name": "I4",
                  "date": "2019-01-18",
                  "description": "M. Bry will give you very interesting tips to build your first business model !",
                  "hour": "16:30",
                  "lesson_number": 1,
                  "location": "Telecom",
                },
                Object {
                  "course_id": 2,
                  "course_short_name": "I4",
                  "date": "2019-01-27",
                  "description": "Nicolas Bry will introduce you to the famous post-it technique",
                  "hour": "16:30",
                  "lesson_number": 2,
                  "location": "Telecom",
                },
                Object {
                  "course_id": 2,
                  "course_short_name": "I4",
                  "date": "2019-01-26",
                  "description": "You will now present your project, even if it is very shitty",
                  "hour": "14:30",
                  "lesson_number": 3,
                  "location": "Telecom",
                },
                Object {
                  "course_id": 2,
                  "course_short_name": "I4",
                  "date": "",
                  "description": "Ce cours n'existe pas",
                  "hour": "9:30",
                  "lesson_number": 10,
                  "location": "Telecom",
                },
              ],
              "main_teacher": "J-M. Dalle",
              "short_name": "I4",
              "title": "Management de l'innovation ",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Course)
          description="This course is the third of the innovation seminary"
          id={1}
          lessons={
                    Array [
                              Object {
                                "course_id": 1,
                                "course_short_name": "SEMI",
                                "date": "2019-01-18",
                                "description": "JMD will be talking about Agoranov for one long hour",
                                "hour": "16:45",
                                "lesson_number": 1,
                                "location": "Agoranov",
                              },
                              Object {
                                "course_id": 1,
                                "course_short_name": "SEMI",
                                "date": undefined,
                                "description": "JB Bengo will be talking about very interesting things",
                                "hour": "15:15",
                                "lesson_number": 2,
                                "location": "Telecom",
                              },
                              Object {
                                "course_id": 1,
                                "course_short_name": "SEMI",
                                "date": "2019-01-18",
                                "description": "JB Bengo will be talking about very interesting things",
                                "hour": "15:18",
                                "lesson_number": 3,
                                "location": "Telecom",
                              },
                              Object {
                                "course_id": 1,
                                "course_short_name": "SEMI",
                                "date": "2019-01-18",
                                "description": "Version2 : JB Bengo will be talking about very interesting things",
                                "hour": "15:18",
                                "lesson_number": 4,
                                "location": "Telecom",
                              },
                            ]
          }
          main_teacher="J-M. Dalle"
          short_name="SEMI"
          showLessons={false}
          title="Seminaire Innovation"
/>,
        <Connect(Course)
          description="This course is the first of the innovation seminary"
          id={2}
          last_modified={0}
          lessons={
                    Array [
                              Object {
                                "course_id": 2,
                                "course_short_name": "I4",
                                "date": "2019-01-18",
                                "description": "M. Bry will give you very interesting tips to build your first business model !",
                                "hour": "16:30",
                                "lesson_number": 1,
                                "location": "Telecom",
                              },
                              Object {
                                "course_id": 2,
                                "course_short_name": "I4",
                                "date": "2019-01-27",
                                "description": "Nicolas Bry will introduce you to the famous post-it technique",
                                "hour": "16:30",
                                "lesson_number": 2,
                                "location": "Telecom",
                              },
                              Object {
                                "course_id": 2,
                                "course_short_name": "I4",
                                "date": "2019-01-26",
                                "description": "You will now present your project, even if it is very shitty",
                                "hour": "14:30",
                                "lesson_number": 3,
                                "location": "Telecom",
                              },
                              Object {
                                "course_id": 2,
                                "course_short_name": "I4",
                                "date": "",
                                "description": "Ce cours n'existe pas",
                                "hour": "9:30",
                                "lesson_number": 10,
                                "location": "Telecom",
                              },
                            ]
          }
          main_teacher="J-M. Dalle"
          short_name="I4"
          showLessons={false}
          title="Management de l'innovation "
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "description": "This course is the third of the innovation seminary",
          "id": 1,
          "lessons": Array [
            Object {
              "course_id": 1,
              "course_short_name": "SEMI",
              "date": "2019-01-18",
              "description": "JMD will be talking about Agoranov for one long hour",
              "hour": "16:45",
              "lesson_number": 1,
              "location": "Agoranov",
            },
            Object {
              "course_id": 1,
              "course_short_name": "SEMI",
              "date": undefined,
              "description": "JB Bengo will be talking about very interesting things",
              "hour": "15:15",
              "lesson_number": 2,
              "location": "Telecom",
            },
            Object {
              "course_id": 1,
              "course_short_name": "SEMI",
              "date": "2019-01-18",
              "description": "JB Bengo will be talking about very interesting things",
              "hour": "15:18",
              "lesson_number": 3,
              "location": "Telecom",
            },
            Object {
              "course_id": 1,
              "course_short_name": "SEMI",
              "date": "2019-01-18",
              "description": "Version2 : JB Bengo will be talking about very interesting things",
              "hour": "15:18",
              "lesson_number": 4,
              "location": "Telecom",
            },
          ],
          "main_teacher": "J-M. Dalle",
          "short_name": "SEMI",
          "showLessons": false,
          "title": "Seminaire Innovation",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "description": "This course is the first of the innovation seminary",
          "id": 2,
          "last_modified": 0,
          "lessons": Array [
            Object {
              "course_id": 2,
              "course_short_name": "I4",
              "date": "2019-01-18",
              "description": "M. Bry will give you very interesting tips to build your first business model !",
              "hour": "16:30",
              "lesson_number": 1,
              "location": "Telecom",
            },
            Object {
              "course_id": 2,
              "course_short_name": "I4",
              "date": "2019-01-27",
              "description": "Nicolas Bry will introduce you to the famous post-it technique",
              "hour": "16:30",
              "lesson_number": 2,
              "location": "Telecom",
            },
            Object {
              "course_id": 2,
              "course_short_name": "I4",
              "date": "2019-01-26",
              "description": "You will now present your project, even if it is very shitty",
              "hour": "14:30",
              "lesson_number": 3,
              "location": "Telecom",
            },
            Object {
              "course_id": 2,
              "course_short_name": "I4",
              "date": "",
              "description": "Ce cours n'existe pas",
              "hour": "9:30",
              "lesson_number": 10,
              "location": "Telecom",
            },
          ],
          "main_teacher": "J-M. Dalle",
          "short_name": "I4",
          "showLessons": false,
          "title": "Management de l'innovation ",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Course)
            description="This course is the third of the innovation seminary"
            id={1}
            lessons={
                        Array [
                                    Object {
                                      "course_id": 1,
                                      "course_short_name": "SEMI",
                                      "date": "2019-01-18",
                                      "description": "JMD will be talking about Agoranov for one long hour",
                                      "hour": "16:45",
                                      "lesson_number": 1,
                                      "location": "Agoranov",
                                    },
                                    Object {
                                      "course_id": 1,
                                      "course_short_name": "SEMI",
                                      "date": undefined,
                                      "description": "JB Bengo will be talking about very interesting things",
                                      "hour": "15:15",
                                      "lesson_number": 2,
                                      "location": "Telecom",
                                    },
                                    Object {
                                      "course_id": 1,
                                      "course_short_name": "SEMI",
                                      "date": "2019-01-18",
                                      "description": "JB Bengo will be talking about very interesting things",
                                      "hour": "15:18",
                                      "lesson_number": 3,
                                      "location": "Telecom",
                                    },
                                    Object {
                                      "course_id": 1,
                                      "course_short_name": "SEMI",
                                      "date": "2019-01-18",
                                      "description": "Version2 : JB Bengo will be talking about very interesting things",
                                      "hour": "15:18",
                                      "lesson_number": 4,
                                      "location": "Telecom",
                                    },
                                  ]
            }
            main_teacher="J-M. Dalle"
            short_name="SEMI"
            showLessons={false}
            title="Seminaire Innovation"
/>,
          <Connect(Course)
            description="This course is the first of the innovation seminary"
            id={2}
            last_modified={0}
            lessons={
                        Array [
                                    Object {
                                      "course_id": 2,
                                      "course_short_name": "I4",
                                      "date": "2019-01-18",
                                      "description": "M. Bry will give you very interesting tips to build your first business model !",
                                      "hour": "16:30",
                                      "lesson_number": 1,
                                      "location": "Telecom",
                                    },
                                    Object {
                                      "course_id": 2,
                                      "course_short_name": "I4",
                                      "date": "2019-01-27",
                                      "description": "Nicolas Bry will introduce you to the famous post-it technique",
                                      "hour": "16:30",
                                      "lesson_number": 2,
                                      "location": "Telecom",
                                    },
                                    Object {
                                      "course_id": 2,
                                      "course_short_name": "I4",
                                      "date": "2019-01-26",
                                      "description": "You will now present your project, even if it is very shitty",
                                      "hour": "14:30",
                                      "lesson_number": 3,
                                      "location": "Telecom",
                                    },
                                    Object {
                                      "course_id": 2,
                                      "course_short_name": "I4",
                                      "date": "",
                                      "description": "Ce cours n'existe pas",
                                      "hour": "9:30",
                                      "lesson_number": 10,
                                      "location": "Telecom",
                                    },
                                  ]
            }
            main_teacher="J-M. Dalle"
            short_name="I4"
            showLessons={false}
            title="Management de l'innovation "
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "description": "This course is the third of the innovation seminary",
            "id": 1,
            "lessons": Array [
              Object {
                "course_id": 1,
                "course_short_name": "SEMI",
                "date": "2019-01-18",
                "description": "JMD will be talking about Agoranov for one long hour",
                "hour": "16:45",
                "lesson_number": 1,
                "location": "Agoranov",
              },
              Object {
                "course_id": 1,
                "course_short_name": "SEMI",
                "date": undefined,
                "description": "JB Bengo will be talking about very interesting things",
                "hour": "15:15",
                "lesson_number": 2,
                "location": "Telecom",
              },
              Object {
                "course_id": 1,
                "course_short_name": "SEMI",
                "date": "2019-01-18",
                "description": "JB Bengo will be talking about very interesting things",
                "hour": "15:18",
                "lesson_number": 3,
                "location": "Telecom",
              },
              Object {
                "course_id": 1,
                "course_short_name": "SEMI",
                "date": "2019-01-18",
                "description": "Version2 : JB Bengo will be talking about very interesting things",
                "hour": "15:18",
                "lesson_number": 4,
                "location": "Telecom",
              },
            ],
            "main_teacher": "J-M. Dalle",
            "short_name": "SEMI",
            "showLessons": false,
            "title": "Seminaire Innovation",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "description": "This course is the first of the innovation seminary",
            "id": 2,
            "last_modified": 0,
            "lessons": Array [
              Object {
                "course_id": 2,
                "course_short_name": "I4",
                "date": "2019-01-18",
                "description": "M. Bry will give you very interesting tips to build your first business model !",
                "hour": "16:30",
                "lesson_number": 1,
                "location": "Telecom",
              },
              Object {
                "course_id": 2,
                "course_short_name": "I4",
                "date": "2019-01-27",
                "description": "Nicolas Bry will introduce you to the famous post-it technique",
                "hour": "16:30",
                "lesson_number": 2,
                "location": "Telecom",
              },
              Object {
                "course_id": 2,
                "course_short_name": "I4",
                "date": "2019-01-26",
                "description": "You will now present your project, even if it is very shitty",
                "hour": "14:30",
                "lesson_number": 3,
                "location": "Telecom",
              },
              Object {
                "course_id": 2,
                "course_short_name": "I4",
                "date": "",
                "description": "Ce cours n'existe pas",
                "hour": "9:30",
                "lesson_number": 10,
                "location": "Telecom",
              },
            ],
            "main_teacher": "J-M. Dalle",
            "short_name": "I4",
            "showLessons": false,
            "title": "Management de l'innovation ",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
